jobs/release.Jenkinsfile

```diff
@@ -152,34 +152,64 @@ lock(resource: "release-${params.STREAM}", extra: locks) {
-            if (oscontainer_registry_repo) {
-                withCredentials([file(variable: 'REGISTRY_SECRET',
-                                      credentialsId: 'oscontainer-push-registry-secret')]) {
-                    def arch_args = basearches.collect{"--arch ${it}"}.join(" ")
-                    shwrap("""
-                    cosa push-container-manifest --auth=\${REGISTRY_SECRET} \
-                        --repo=${oscontainer_registry_repo} --tag=${params.STREAM} \
-                        --artifact=ostree --metajsonname=base-oscontainer \
-                        --build=${params.VERSION} ${arch_args}
-                    """)
-                }
-            }
-            def oscontainer_old_registry_repo = pipecfg.registry_repos?.oscontainer_old
-            if (oscontainer_old_registry_repo) {
-                // For a period of time let's also mirror into the old location too
-                // Drop this after October 2022. See
-                // https://discussion.fedoraproject.org/t/updated-registry-location-for-fedora-coreos-ostree-native-container/42740
-                withCredentials([file(credentialsId: 'oscontainer-secret', variable: 'REGISTRY_SECRET')]) {
-                    shwrap("""
-                    skopeo copy --all --authfile \$REGISTRY_SECRET \
-                        docker://${oscontainer_registry_repo}:${params.STREAM} \
-                        docker://${oscontainer_old_registry_repo}:${params.STREAM}
-                    """)
-                }
-            }
+        def registry_repos = pipeutils.get_registry_repos(pipecfg, params.STREAM)
+
+        // [config.yaml name -> [meta.json 'images' name, meta.json toplevel name, tag suffix]]
+        def push_containers = ['oscontainer': ['ostree', 'base-oscontainer', ''],
+                               'extensions': ['extensions-container', 'extensions-container', 'extensions'],
+                               'legacy_oscontainer': ['ostree', 'oscontainer', 'legacy']]
+
+        // filter out those not defined in the config
+        push_containers.keySet().retainAll(registry_repos.keySet())
+
+        // filter out those not built. this step makes the assumption that if an
+        // image isn't built for x86_64, then we don't upload it at all
+        def artifacts = readJSON(text: shwrapCapture("cosa meta --dump")).images.keySet()
+        // in newer Groovy, retainAll can take a closure, which would be nicer here
+        for (key in (push_containers.keySet() as List)) {
+          if (!(push_containers[key][0] in artifacts)) {
+            push_containers.remove(key)
+          }
+        }
+
+        if (push_containers) {
+          stage("Push Containers") {
+            parallel container_types.collectEntries{configname, val -> [key, {
+              withCredentials([file(variable: 'REGISTRY_SECRET',
+                                    credentialsId: 'oscontainer-push-registry-secret')]) {
```

Comment:
assuming here that we'll be using the same secret for all probably won't hold forever. I guess if it is true now then let's go with it.

It also is a little weird to call it the `oscontainer-push-registry-secret` when we have extensions in there too.

Reply:
Yup, agreed. I'd rather not add support for it until we actually need it.

Re. naming, let's keep that for a follow-up?

Reply:
> Re. naming, let's keep that for a follow-up?

That's fair, but note we will have to updated the existing `oscontainer-secret` in our pipeline if this merges as is.

Reply:
Yes indeed. Planning to coordinate that with the merge.